etl/src/v2/replication/apply.rs

```diff
@@ -145,10 +145,41 @@ impl StatusUpdate {
     }
 }
 
-#[derive(Debug, Clone)]
-enum BatchEarlyBreak {
-    Break,
-    BreakAndDiscard,
+/// An enum representing if the batch should be ended or not
+#[derive(Debug)]
+enum EndBatch {
+    /// The batch should include the last processed event and end.
+    Inclusive,
+
+    /// The batch should exclude the last processed event and end.
+    Exclusive,
+}
+
+/// Result returned from `handle_replication_message` and related functions
+#[derive(Debug, Default)]
+struct HandleMessageResult {
+    /// The event converted from the replication message.
+    /// Could be None if this event should not be added to the batch
+    /// Will be None in the following cases:
+    ///
+    /// * When the apply worker receives an event for a table which is not ready
+    /// * When the apply or table sync workers receive an event from a table which was skipped
+    /// * When the table sync worker receives an event from a table other than its own
+    /// * When the message is a primary keepalive message
+    ///
+    event: Option<Event>,
+
+    /// Set to a commit message's end_lsn value, None otherwise
+    end_lsn: Option<PgLsn>,
+
+    /// Set when a batch should be ended. Can be set to Inclusive when a commit
+    /// message indicates that it will mark the table sync worker as caught up.
+    /// Can be set to Exclusive when a replication message indicates a change in
+    /// schema
```

Comment:
nit: I would explain this a bit better and also that those are just examples where the `EndBatch` is used.